### ISSUE TITLE
Latest release has broken everything 😧

### ISSUE THREAD

**Issue by cperezabo** [idc101/git-mkver#36]
Hi Iain, I think you should take more care about the decisions you make, or at least to communicate them with time to the people working with `git-mkver`.

[setup-git-mkver](https://github.com/marketplace/actions/setup-git-mkver) is broken at the moment, as well as every pipeline using it because v1.3.0 changed the name format of the releases.

The workaround is to stick to `v1.2.2`, but that implies to update all the pipelines that are using `latest`

I'm working to support the new assets name format, but I am doing reverse engineering to the universal package because there is no documentation about how to making it run in Windows globally through `git mkver`, the `/etc/scoop/git-mkver.json` is outdated as well as `/etc/shell/install.sh` so I'm blind.

As for Linux, the new x86_64 binary is failing with the following error in the same ubuntu-20.04 runner in which git-mkver v1.2.2 used to run ok.

```
/usr/local/bin/git-mkver: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/git-mkver)
/usr/local/bin/git-mkver: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/git-mkver)
```

**Comment by idc101** [comment]
Hi Cristián,

Sorry about this! I didn't realise you had made setup-git-mkver (very cool btw) and this would break it.

I'll investigate the linux binary issue and then upload binaries with the old name formats. I don't have a handy windows pc at the moment (one of the reasons I built the universal version), but let me know how I can help.

Iain

**Comment by cperezabo** [comment]
Maybe you forgot but you included it at https://idc101.github.io/git-mkver/installation through https://github.com/idc101/git-mkver/issues/16 🤣

I don't have a Windows machine neither, that's why I depend on CI to build Windows binaries.

I am working against the clock to make it work using the universal version. The bash script included there intended to work both in macOS and Linux right?

**Comment by idc101** [comment]
> Maybe you forgot but you included it at https://idc101.github.io/git-mkver/installation through https://github.com/idc101/git-mkver/issues/16 🤣
Oops!

yes script in universal should work on both.

**Comment by idc101** [comment]
v1.2.2 was built on ubuntu:20.04, v1.3.0 now uses 22.04 (from your PR :-) ) and this causes the glibc errors I think.

Would it help if I rebuild using ubuntu:20.04?

**Comment by cperezabo** [comment]
I am trying to make it work with the universal script, give me some minutes and I will tell you what happened

**Comment by cperezabo** [comment]
Well, I was almost able to seamlessly make versions <1.3.0 work the same way versions >=1.3.0 do.

What I am doing is to add a special OS-based path so the system can find git-mkver executable as you can see here https://github.com/cperezabo/setup-git-mkver/blob/next/action.yml#L19-L25

Here you can see how it's being downloaded and installed https://github.com/cperezabo/setup-git-mkver/blob/next/install.sh

Tests in Linux are OK, as for Windows they are failing as you can see in https://github.com/cperezabo/setup-git-mkver/actions/runs/4802013298

The problem in Windows is that now the executable is a .bat file, and I think git is not recognizing it the same way it did it with `git-mkver.exe`, so `git mkver` is not working as you can see here https://github.com/cperezabo/setup-git-mkver/actions/runs/4802013298/jobs/8544950547

**Comment by cperezabo** [comment]
If you ask yourself why I am using System32 folder in Windows, it's just because prior 1.3.0 I was placing the git-mkver binary in that folder, and as for Linux I was using /usr/local/bin, I could change it after making it work

**Comment by idc101** [comment]
Ok I see what you mean about the `.bat` not being recognised as an executable. I will setup a GitHub Windows Runner to build the windows binary.

**Comment by idc101** [comment]
@cperezabo I've uploaded a native windows binary to the 1.3.0 release. Hope that is useful.

**Comment by cperezabo** [comment]
Hi, thanks, but could you keep the arch naming? previously you were using amd64, is there a reason for now using x64 instead? I'm asking just to simplify the script.

**Comment by idc101** [comment]
Uploaded a version with the old naming conventions for windows. Not sure on the best naming. Ideally an equivalent of `arch` linux command on windows.

**Comment by cperezabo** [comment]
I changed the script, but it failed again, and the problem is that the .exe inside the Windows zip file is not named "git-mkver.exe", as it was named before.

**Comment by cperezabo** [comment]
You can see [here](https://github.com/cperezabo/setup-git-mkver/actions/runs/4822392423/jobs/8589566104) that the first tests with a previous version works, but using the latest fails

**Comment by idc101** [comment]
Ok, uploaded it again, changing the exe name.

**Comment by cperezabo** [comment]
It worked but something strange is happening, I am using `windows-2019`, and I noticed you used `windows-latest` so I changed it to `windows-latest` as well, and running `git mkver` fails with a `Process completed with exit code 53.`. It makes no sense. Any idea? [Here](https://github.com/cperezabo/setup-git-mkver/actions/runs/4822794776) you can see the last integration

**Comment by cperezabo** [comment]
You won't see the error 53 there because I tried it in another run [here](https://github.com/cperezabo/setup-git-mkver/actions/runs/4822697968/jobs/8590243146), when it failed the first time, I tried to execute git mkver directly and there I saw it.

**Comment by idc101** [comment]
I've run the binary on another windows laptop with no issues. What is interesting is that v1.2.2 fails with error code 53 and that binary hasn't changed.

**Comment by cperezabo** [comment]
Sorry, my fault! I am already dizzy with so much round and round, what is failing is v1.2.2! In fact v1.3.0 is not being test because v1.2.2 test fails first. Well, I think maybe v1.2.2 should be compiled in windows-latest as well, which I don't think it's worth it. Maybe I can just tests v1.3.0 or higher in windows latest and v1.2.2 or lower in windows 2019 only. What do you think? It should be documented somewhere that v1.2.2 will not work in windows-2022 btw.

**Comment by cperezabo** [comment]
Well, I released setup-git-mkver@v1.2.0 because I was thinking about it, and the fact that older Windows binaries won't work in windows-2022 is out of the scope of setup-git-mkver, the instalation process is the same.
For linux it is using the universal version.
